tests/pytests/unit/proxy/test_netmiko_px.py

```diff
@@ -0,0 +1,220 @@
+        random_string("proxy-minion-"),
+    )
+    return factory.config
+
+
+@pytest.fixture
+def proxy_minion_config(proxy_minion_config_module):
+    return copy.deepcopy(proxy_minion_config_module)
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {netmiko_proxy: {}}
+
+
+class MockNetmikoConnection:
+    def is_alive(self):
+        return False
+
+    def send_config_set(self, *args, **kwargs):
+        return args, kwargs
+
+
+def test_check_virtual():
+    """
+    check netmiko_proxy virtual method - return value
+    """
+
+    with patch.object(netmiko_proxy, "HAS_NETMIKO", True):
+        result = netmiko_proxy.__virtual__()
+        assert "netmiko" in result
+
+    expected = (
+        False,
+        "The netmiko proxy module requires netmiko library to be installed.",
+    )
+    with patch.object(netmiko_proxy, "HAS_NETMIKO", False):
+        result = netmiko_proxy.__virtual__()
+        assert expected == result
+
+
+def test_init_skip_connect_on_init_true(proxy_minion_config):
+    """
+    check netmiko_proxy init method
+    """
+
+    proxy_minion_config["skip_connect_on_init"] = True
+
+    result = netmiko_proxy.init(proxy_minion_config)
+    assert "netmiko_device" in netmiko_proxy.__context__
+    assert "args" in netmiko_proxy.__context__["netmiko_device"]
+
+    assert "initialized" in netmiko_proxy.__context__["netmiko_device"]
+    assert not netmiko_proxy.__context__["netmiko_device"]["initialized"]
+
+    assert "up" in netmiko_proxy.__context__["netmiko_device"]
+    assert netmiko_proxy.__context__["netmiko_device"]["up"]
+
+    assert "always_alive" in netmiko_proxy.__context__["netmiko_device"]
+
+    assert "conneciton" not in netmiko_proxy.__context__["netmiko_device"]
+
+
+def test_init_skip_connect_on_init_false(proxy_minion_config):
+    """
+    check netmiko_proxy init method
+    """
+
+    proxy_minion_config["skip_connect_on_init"] = False
+
+    mock_make_con = MagicMock()
+    with patch.object(netmiko_proxy, "make_con", mock_make_con):
+        result = netmiko_proxy.init(proxy_minion_config)
+
+    assert "netmiko_device" in netmiko_proxy.__context__
+    assert "args" in netmiko_proxy.__context__["netmiko_device"]
+
+    assert "initialized" in netmiko_proxy.__context__["netmiko_device"]
+    assert netmiko_proxy.__context__["netmiko_device"]["initialized"]
+
+    assert "up" in netmiko_proxy.__context__["netmiko_device"]
+    assert netmiko_proxy.__context__["netmiko_device"]["up"]
+
+    assert "always_alive" in netmiko_proxy.__context__["netmiko_device"]
+
+    assert "conneciton" not in netmiko_proxy.__context__["netmiko_device"]
+
+
+def test_make_con(proxy_minion_config):
+    """
+    check netmiko_proxy make_con method
+    """
+
+    proxy_minion_config["skip_connect_on_init"] = True
+
+    netmiko_proxy.init(proxy_minion_config)
+
+    mock_connection = MockNetmikoConnection
+
+    with patch.object(netmiko_proxy, "ConnectHandler", mock_connection, create=True):
```

Comment:
**suggestion** This is mocked several places, prefer extracting to a fixture. Also note, no class definition is required:

```
@pytest.fixture
def fake_connection_handler():
    with patch.object(netmiko_proxy, "ConnectHandler", create=True) as  fake_handler:
        fake_handler.is_alive.return_value = True
        fake_handler.send_config_set.side_effect = lambda *args, **kwargs: args, kwargs
        yield # fake_handler is optional
```

This will simplify many of these tests.

Reply:
I only migrated the test because it used the config file which was deleted.
Let's do your suggestion on another PR?

Reply:
Actually, only one test could use that fixture, other tests that patch `netmiko_proxy.ConnectHandler` do not share the same patch parameters...